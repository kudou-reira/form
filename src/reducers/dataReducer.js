import data from '../lib/dishes.json';

export const initialState = {
	data: data
};

export default (state = initialState) => {
	return state;
}
