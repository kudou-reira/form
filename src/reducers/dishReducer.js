import {
  DISH_COLLECTION_UPDATE,
  ADD_DISH,
  DELETE_DISH,
  RECORD_DISHES,
  DISH_COLLECTION_RESET,
  VERIFY_DISHES
} from '../actions/types';

const INITIAL_STATE = {
	dishCollection: [],
	verifyDishes: false,
	recordDishes: 0
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
  	case ADD_DISH:
  		return {...state, dishCollection: [...state.dishCollection, action.payload]};
  	case DELETE_DISH:
  		let index = state.dishCollection.findIndex((dish) => dish.id === action.payload); 
  		return {...state, dishCollection: [...state.dishCollection.slice(0, index), ...state.dishCollection.slice(index+1)]};
  	case RECORD_DISHES:
  		return {...state, recordDishes: action.payload}
  	case VERIFY_DISHES:
      return {...state, verifyDishes: action.payload}
  	case DISH_COLLECTION_RESET:
  		return {...state, dishCollection: []}
  	case DISH_COLLECTION_UPDATE:
  		var eventKey = action.payload.eventKey;
  		var type = typeof(eventKey);
  		var id = action.payload.id;
  		const existsAt = state.dishCollection.findIndex(dish => dish.id === id);
  		if(existsAt === -1) {
  			if(type === "string") {
  				var tempObject = {
  					id: id,
  					dish: eventKey,
  					servings: 1
  				}
  				return {...state, dishCollection: [...state.dishCollection, tempObject]};
  			}
  		} else {
  			console.log("this is existsAt", existsAt);
  			// exists in the array
  			// process update
  			// !! IMPORTANT !!
  			// check if the eventKey is a string and if it's equal to the current one or not
  			if(type === "string") {
  				console.log("this is type", type);
					return {
		        ...state,
		        dishCollection: state.dishCollection.map(dish => dish.id === id ?
	            // transform the one with a matching id
	            { ...dish, dish: eventKey } : 
	            // otherwise return original dish
	            dish
		        ) 
			    };
  			} else if(type === "number") {
  				console.log("this is type", type);
					return {
		        ...state,
		        dishCollection: state.dishCollection.map(dish => dish.id === id ?
	            // transform the one with a matching id
	            { ...dish, servings: eventKey } : 
	            // otherwise return original dish
	            dish
		        ) 
			    };
  			}
  		}
    default:
      return state;
  }
}
