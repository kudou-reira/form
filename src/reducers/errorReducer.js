import {
  ERROR,
  ERROR_SERVING
} from '../actions/types';

export const initialState = {
	error: true,
	errorServing: true
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case ERROR:
  		return {...state, error: action.payload}
  	case ERROR_SERVING:
  		return {...state, errorServing: action.payload}
    default:
      return state;
  }
}
