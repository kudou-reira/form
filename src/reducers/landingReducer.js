import {
  FETCH_MEALTIME
} from '../actions/types';

const INITIAL_STATE = {
	mealtime: ''
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
  	case FETCH_MEALTIME:
  		return {...state, mealtime: action.payload}
    default:
      return state;
  }
}
