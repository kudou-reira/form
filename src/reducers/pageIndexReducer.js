import {
  PAGE_INDEX
} from '../actions/types';

export const initialState = {
	page: 1
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case PAGE_INDEX:
  		return {...state, page: action.payload}
    default:
      return state;
  }
}
