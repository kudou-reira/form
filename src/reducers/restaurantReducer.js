import {
  SELECT_RESTAURANT,
  VERIFY_RESTAURANT
} from '../actions/types';

export const initialState = {
	restaurant: '----',
  verifyRestaurant: false
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case SELECT_RESTAURANT:
  		return {...state, restaurant: action.payload}
    case VERIFY_RESTAURANT:
      return {...state, verifyRestaurant: action.payload}
    default:
      return state;
  }
}
