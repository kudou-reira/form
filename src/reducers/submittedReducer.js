import {
  VERIFY_SUBMITTED
} from '../actions/types';

export const initialState = {
	verifySubmitted: false
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case VERIFY_SUBMITTED:
      return {...state, verifySubmitted: action.payload}
    default:
      return state;
  }
}
