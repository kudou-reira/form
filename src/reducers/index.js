import { combineReducers } from 'redux';
import landingReducer from './landingReducer';
import restaurantReducer from './restaurantReducer';
import pageIndexReducer from './pageIndexReducer';
import dataReducer from './dataReducer';
import dishReducer from './dishReducer';
import errorReducer from './errorReducer';

export default combineReducers({
	landing: landingReducer,
	restaurant: restaurantReducer,
	pageIndex: pageIndexReducer,
	data: dataReducer,
	dish: dishReducer,
	error: errorReducer
});

