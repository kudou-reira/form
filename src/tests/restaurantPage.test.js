import RestaurantPage from '../components/restaurantPage';
import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { initialState as data } from '../reducers/dataReducer';
import { initialState as error } from '../reducers/errorReducer';
import { initialState as restaurant } from '../reducers/restaurantReducer';
import { initialState as landing } from '../reducers/landingReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<RestaurantPage />', () => {
  it('renders with default props', () => {
  	const store = mockStore({landing, restaurant, error, data});
    const wrapper = shallow(
      <RestaurantPage store={store} />
    );
    // console.log("this is wrapper props in first test", wrapper.props());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
