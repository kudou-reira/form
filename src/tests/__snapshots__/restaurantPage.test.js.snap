// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantPage /> renders with default props 1`] = `
<RestaurantPage
  addDish={[Function]}
  data={
    Object {
      "data": Object {
        "dishes": Array [
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 1,
            "name": "Chicken Burger",
            "restaurant": "Mc Donalds",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 2,
            "name": "Ham Burger",
            "restaurant": "Mc Donalds",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 3,
            "name": "Cheese Burger",
            "restaurant": "Mc Donalds",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 4,
            "name": "Fries",
            "restaurant": "Mc Donalds",
          },
          Object {
            "availableMeals": Array [
              "breakfast",
            ],
            "id": 5,
            "name": "Egg Muffin",
            "restaurant": "Mc Donalds",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 6,
            "name": "Burrito",
            "restaurant": "Taco Bell",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 7,
            "name": "Tacos",
            "restaurant": "Taco Bell",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 8,
            "name": "Quesadilla",
            "restaurant": "Taco Bell",
          },
          Object {
            "availableMeals": Array [
              "dinner",
            ],
            "id": 9,
            "name": "Steak",
            "restaurant": "BBQ Hut",
          },
          Object {
            "availableMeals": Array [
              "dinner",
            ],
            "id": 10,
            "name": "Yakitori",
            "restaurant": "BBQ Hut",
          },
          Object {
            "availableMeals": Array [
              "dinner",
            ],
            "id": 11,
            "name": "Nankotsu",
            "restaurant": "BBQ Hut",
          },
          Object {
            "availableMeals": Array [
              "dinner",
            ],
            "id": 12,
            "name": "Piman",
            "restaurant": "BBQ Hut",
          },
          Object {
            "availableMeals": Array [
              "lunch",
            ],
            "id": 13,
            "name": "Vegan Bento",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "breakfast",
            ],
            "id": 14,
            "name": "Coleslaw Sandwich",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "breakfast",
            ],
            "id": 15,
            "name": "Grilled Sandwich",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 16,
            "name": "Veg. Salad",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 17,
            "name": "Fruit Salad",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 18,
            "name": "Corn Soup",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 19,
            "name": "Tomato Soup",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 20,
            "name": "Minestrone Soup",
            "restaurant": "Vege Deli",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 21,
            "name": "Pepperoni Pizza",
            "restaurant": "Pizzeria",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 22,
            "name": "Pepperoni Pizza",
            "restaurant": "Pizzeria",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 23,
            "name": "Hawaiian Pizza",
            "restaurant": "Pizzeria",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 24,
            "name": "Seafood Pizza",
            "restaurant": "Pizzeria",
          },
          Object {
            "availableMeals": Array [
              "dinner",
            ],
            "id": 25,
            "name": "Deep Dish Pizza",
            "restaurant": "Pizzeria",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 26,
            "name": "Chow Mein",
            "restaurant": "Panda Express",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 27,
            "name": "Mapo Tofu",
            "restaurant": "Panda Express",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 28,
            "name": "Kung Pao",
            "restaurant": "Panda Express",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 29,
            "name": "Wontons",
            "restaurant": "Panda Express",
          },
          Object {
            "availableMeals": Array [
              "breakfast",
              "lunch",
              "dinner",
            ],
            "id": 30,
            "name": "Garlic Bread",
            "restaurant": "Olive Garden",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 31,
            "name": "Ravioli",
            "restaurant": "Olive Garden",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 32,
            "name": "Rigatoni Spaghetti",
            "restaurant": "Olive Garden",
          },
          Object {
            "availableMeals": Array [
              "lunch",
              "dinner",
            ],
            "id": 33,
            "name": "Fettucine Pasta",
            "restaurant": "Olive Garden",
          },
        ],
      },
    }
  }
  deleteDish={[Function]}
  dishCollectionReset={[Function]}
  dishCollectionUpdate={[Function]}
  error={
    Object {
      "error": true,
      "errorServing": true,
    }
  }
  landing={
    Object {
      "mealtime": "----",
      "numberOfPeople": "----",
      "verifyLanding": false,
    }
  }
  recordDishes={[Function]}
  restaurant={
    Object {
      "restaurant": "----",
      "verifyRestaurant": false,
    }
  }
  selectMealtime={[Function]}
  selectPeople={[Function]}
  selectRestaurant={[Function]}
  sendError={[Function]}
  sendErrorServing={[Function]}
  sendPageIndex={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  verifyDishes={[Function]}
  verifyLanding={[Function]}
  verifyRestaurant={[Function]}
  verifySubmitted={[Function]}
/>
`;
